### DISCUSSION TITLE
How to insert a multi row on the same transaction?

### DISCUSSION THREAD

**Discussion by glaucioguerra** [sijms/go-ora#163] [General]
Hi,

I'm trying to insert a lot of rows using the same transaction, trying to improve my bulk insert. What is the best option to do that?

I know that we can use the INSERT ALL INTO statement on the Oracle database, but I'm not sure if this is the best way.

Thanks!

**Comment by sijms** [comment]
I am working now on bulk insert function insert multiple rows in one command

**Reply by glaucioguerra** [reply]
That's amazing @sijms !

**Comment by sijms** [comment]
hi @glaucioguerra 
I make new release with requested feature
now you can insert large data with one network call which is dramatically improve the performance
you can see an example of this feature compared to traditional insert
if you need transaction you can use sql.Tx

**Reply by glaucioguerra** [reply]
Hi Samy,

Thanks a lot. I will test tomorrow and feedback on the performance improvements. 

Thanks again!

**Reply by glaucioguerra** [reply]
Hi @sijms,

In my use case, my performance (using database on Internet) was something like this:

8747 lines
4m 5s processing time without the bulk insert.

![image](https://user-images.githubusercontent.com/122855/155251649-35624156-0389-4426-997b-b86127986a5c.png)


8747 lines
2.7s with bulk insert.

![image](https://user-images.githubusercontent.com/122855/155251683-477cce16-1853-4984-85fc-bd03be84412a.png)


That's amazing! Thanks again.
